### PR TITLE
integrate proofs in the documentation site

### LINKED ISSUE #56: Integrate proofs on Transformations/Measurements
The exact route we go here is still pretty open-ended.

- Write latex documents on overleaf.com and add links to them from the documentation for each struct.
- Write latex documents and check them in on this repository, link to this repository from the documentation for each struct.
- Other?

Additional questions-
Do we provide attribution for who has written the proof for the relation? 
How do we provide attribution for who has written the proof for the relation?
How do we record vetting on the documents/code?

### PR DESCRIPTION
**Pull request by Shoeboxam**
Closes #56 

* updated docs.yml workflow to generate all proof pdfs and upload to `opendp/artifacts` upon merge to main or release
* updated docs.yml to copy proof artifacts to gh-pages docs site's multiversion folders
* source code links in docs.opendp.org for automatically generated functions go to the same function in docs.rs
* support for rendering katex in the rustdocs
    * added an `opendp_tooling/katex.html` that replaces latex in $inline$ and ```math sections
    * updated developer documentation to inject the html file
    * a VSCode task for building rustdocs with latex
    * added a Cargo.toml directive for docs.rs to render with katex injected
* renamed mod.sty to lib.sty, as this style file sits in the root next to lib.rs
* latex macro for generating a link to either pdf file or rust doc
    * this is used in the `make_count.tex` PR, check out the rendered pdf
* adds OPENDP_SPHINX_PORT and OPENDP_RUSTDOC_PORT for generating links to local docs servers
    * update lib.sty \rustdoc to link local docs server if OPENDP_RUSTDOC_PORT set
    * update documentation for local docs builds
    * adds docs/Makefile tasks for building all latex, building rustdoc, and starting docs servers

This should give us all the pieces we need to continuously browse proofs/documentation across python, rust and latex, with versioned links and support for offline builds.

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -34,6 +34,9 @@ html:
 	@echo
 	@echo "Build finished. The HTML pages are in $(BUILDDIR)/html."
 
+html-rustdoc:
+	cd ../rust && cargo rustdoc --features untrusted,derive -- --html-in-header opendp_tooling/katex.html --document-private-items
```
Comment (andrewvyrros) [inline docs/Makefile:38]: Don't you already specify the katex thing in `Cargo.toml`?

**Review by andrewvyrros — COMMENTED** [review]
Another big milestone!

File: docs/Makefile (hunk 1 above)
Comment (Shoeboxam) [reply docs/Makefile:38]: That makes it the html apply when rendered on docs.rs only, unfortunately.

File: rust/Cargo.toml [hunk 2]
```diff
@@ -81,6 +81,10 @@ crate-type = ["rlib", "cdylib"]
 
 [package.metadata.docs.rs]
 features = ["untrusted", "ffi"]
+rustdoc-args = [
+    "--html-in-header",
+    ".cargo/registry/src/github.com-1ecc6299db9ec823/opendp_tooling-0.6.0/katex.html",
```
[the remark this answers is not in the document]
Comment (Shoeboxam) [reply rust/Cargo.toml:87]: I don't think that hash changes. In fact, you should have it on your filesystem too at `~/.cargo/registry/src/github.com-1ecc6299db9ec823/`.

[the remark this answers is not in the document]
Comment (Shoeboxam) [reply rust/Cargo.toml:87]: The hash is computed here: https://github.com/rust-lang/cargo/blob/c5679e21ef8c7deabce3911ce04ead338ff063be/src/cargo/core/source.rs#L437-L450

[the remark this answers is not in the document]
Comment (andrewvyrros) [reply rust/Cargo.toml:87]: Thanks for explaining!

File: docs/Makefile (hunk 1 above)
Comment (andrewvyrros) [reply docs/Makefile:38]: I missed that, thanks!

**Review by andrewvyrros — APPROVED** [review]

**Pull request merged**